k2/torch/csrc/fsa_class_test.cu

```diff
@@ -74,44 +73,42 @@ TEST(RaggedArcTest, FromUnaryFunctionTensor) {
                       torch::dtype(torch::kFloat32).device(device))));
 
     EXPECT_TRUE(torch::equal(
-        dest.GetAttr("int_attr").cast<torch::Tensor>(),
+        dest.GetAttr("int_attr").toTensor(),
         torch::tensor({2, 1, 3}, torch::dtype(torch::kInt32).device(device))));
 
     RaggedAny expected_ragged_attr =
-        RaggedAny("[[5 6] [1 2 3] []]", py::cast(torch::kInt32), device);
-    EXPECT_EQ(dest.GetAttr("ragged_attr").cast<RaggedAny>().ToString(true),
+        RaggedAny("[[5 6] [1 2 3] []]", torch::kInt32, device);
+
+    EXPECT_EQ(ToRaggedAny(dest.GetAttr("ragged_attr")).ToString(true),
               expected_ragged_attr.ToString(true));
 
-    EXPECT_EQ(dest.GetAttr("attr1").cast<std::string>(),
-              src.GetAttr("attr1").cast<std::string>());
+    EXPECT_EQ(dest.GetAttr("attr1").toStringRef(),
+              src.GetAttr("attr1").toStringRef());
 
-    EXPECT_EQ(dest.GetAttr("attr2").cast<std::string>(),
-              src.GetAttr("attr2").cast<std::string>());
+    EXPECT_EQ(dest.GetAttr("attr2").toStringRef(),
+              src.GetAttr("attr2").toStringRef());
 
     torch::Tensor scale = torch::tensor(
         {10, 20, 30}, torch::dtype(torch::kFloat32).device(device));
 
     torch::Tensor sum_attr =
-        (dest.GetAttr("float_attr").cast<torch::Tensor>() * scale).sum();
+        (dest.GetAttr("float_attr").toTensor() * scale).sum();
     torch::Tensor sum_score = (dest.Scores() * scale).sum();
 
-    {
-      py::gil_scoped_release no_gil;
-      torch::autograd::backward({sum_attr}, {});
-      torch::autograd::backward({sum_score}, {});
-    }
+    torch::autograd::backward({sum_attr}, {});
```

Comment:
Can you use
```cpp
sum_attr.backward();
sum_score.backward();
```
?

It's more intuitive and similar to what we are doing in PyTorch.